File: backend/app/schemas/branches.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class BranchCreate(BaseModel):
    code: str
    name: str
    address: str | None = None
    phone: str | None = None


class BranchUpdate(BaseModel):
    code: str | None = None
    name: str | None = None
    address: str | None = None
    phone: str | None = None
    is_active: bool | None = None


class BranchResponse(BaseModel):
    id: UUID
    code: str
    name: str
    address: str | None = None
    phone: str | None = None
    is_active: bool
    created_at: datetime | None = None
    updated_at: datetime | None = None

    model_config = {"from_attributes": True}

File: frontend/types/branches.ts
export type Branch = {
  id: string;
  code: string;
  name: string;
  address?: string | null;
  phone?: string | null;
  is_active: boolean;
  created_at?: string | null;
  updated_at?: string | null;
};

export type BranchCreate = {
  code: string;
  name: string;
  address?: string;
  phone?: string;
};

export type BranchUpdate = {
  code?: string;
  name?: string;
  address?: string;
  phone?: string;
  is_active?: boolean;
};
